### ISSUE TITLE
in_tail: when corrupted log file - situation not reported, not recovered

### ISSUE THREAD

**Issue by epcim** [fluent/fluent-bit#461]
Parsing stuck on this corrupted file.
The expected behaviour would be to report an error about a stuck in_tail process. On the picture is the log in vim editor. The chars is kind of replacement for "long line". After this kind of random data as other log lines without anny issues.

We should detect, that the log increase in the size but we got stuck in some position.

![image](https://user-images.githubusercontent.com/1211552/33494617-160c4474-d6c4-11e7-90bc-d2fba237a8f8.png)

**Comment by epcim** [comment]
2017-10-26 17:20:27 is the last read record, then is the corruption in file (that grep don't have problem to pass

```
ROT2: root@d98-f2-b3-c7-2f-21 (database1):/etc/td-agent-bit # egrep -A5 '2017-10-26 17:20:27' -A 5 /var/log/mysql/mysql_error.log.xz.wrong
2017-10-26 17:20:27 140643391239936 [Note] WSREP: (eb0e6410, 'tcp://0.0.0.0:4567') connection to peer 00000000 with addr tcp://10.244.90.110:4567 timed out, no messages seen in PT3S
171027 07:15:51 mysqld_safe Starting mysqld daemon with databases from /var/lib/mysql
171027 07:15:51 mysqld_safe WSREP: Running position recovery with --log_error='/var/lib/mysql/wsrep_recovery.asfDZW' --pid-file='/var/lib/mysql/d98-f2-b3-c7-2f-21-recover.pid'
2017-10-27  7:15:52 140479553657088 [Note] /usr/sbin/mysqld (mysqld 10.1.25-MariaDB) starting as process 50348 ...
171027 07:16:23 mysqld_safe WSREP: Recovered position 9b56b76c-ba5a-11e7-b4cb-cbba0b087868:420

```

**Comment by epcim** [comment]
This way you may easily decode:
```
 cat /var/log/mysql/mysql_error.log.xz.wrong.b64 | base64
MjAxNy0xMC0yNiAxNzoyMDoyNSAxNDA2NDMzOTEyMzk5MzYgW05vdGVdIFdTUkVQOiAoZWIwZTY0
MTAsICd0Y3A6Ly8wLjAuMC4wOjQ1NjcnKSBjb25uZWN0aW9uIHRvIHBlZXIgMDAwMDAwMDAgd2l0
aCBhZGRyIHRjcDovLzEwLjI0NC45MC44Mjo0NTY3IHRpbWVkIG91dCwgbm8gbWVzc2FnZXMgc2Vl
biBpbiBQVDNTCjIwMTctMTAtMjYgMTc6MjA6MjcgMTQwNjQzMzkxMjM5OTM2IFtOb3RlXSBXU1JF
UDogKGViMGU2NDEwLCAndGNwOi8vMC4wLjAuMDo0NTY3JykgY29ubmVjdGlvbiB0byBwZWVyIDAw
MDAwMDAwIHdpdGggYWRkciB0Y3A6Ly8xMC4yNDQuOTAuMTEwOjQ1NjcgdGltZWQgb3V0LCBubyBt
ZXNzYWdlcyBzZWVuIGluIFBUM1MKAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
AAAAAAAAAAAAADE3MTAyNyAwNzoxNTo1MSBteXNxbGRfc2FmZSBTdGFydGluZyBteXNxbGQgZGFl
bW9uIHdpdGggZGF0YWJhc2VzIGZyb20gL3Zhci9saWIvbXlzcWwKMTcxMDI3IDA3OjE1OjUxIG15
c3FsZF9zYWZlIFdTUkVQOiBSdW5uaW5nIHBvc2l0aW9uIHJlY292ZXJ5IHdpdGggLS1sb2dfZXJy
b3I9Jy92YXIvbGliL215c3FsL3dzcmVwX3JlY292ZXJ5LmFzZkRaVycgLS1waWQtZmlsZT0nL3Zh
ci9saWIvbXlzcWwvZDk4LWYyLWIzLWM3LTJmLTIxLXJlY292ZXIucGlkJwoyMDE3LTEwLTI3ICA3
OjE1OjUyIDE0MDQ3OTU1MzY1NzA4OCBbTm90ZV0gL3Vzci9zYmluL215c3FsZCAobXlzcWxkIDEw
LjEuMjUtTWFyaWFEQikgc3RhcnRpbmcgYXMgcHJvY2VzcyA1MDM0OCAuLi4KMTcxMDI3IDA3OjE2
OjIzIG15c3FsZF9zYWZlIFdTUkVQOiBSZWNvdmVyZWQgcG9zaXRpb24gOWI1NmI3NmMtYmE1YS0x
MWU3LWI0Y2ItY2JiYTBiMDg3ODY4OjQyMAoyMDE3LTEwLTI3ICA3OjE2OjIzIDEzOTk5NDA0NzM1
NTEzNiBbTm90ZV0gL3Vzci9zYmluL215c3FsZCAobXlzcWxkIDEwLjEuMjUtTWFyaWFEQikgc3Rh
cnRpbmcgYXMgcHJvY2VzcyA1MDYzOCAuLi4KMjAxNy0xMC0yNyAgNzoxNjoyMyAxMzk5OTQwNDcz
NTUxMzYgW05vdGVdIFdTUkVQOiBSZWFkIG5pbCBYSUQgZnJvbSBzdG9yYWdlIGVuZ2luZXMsIHNr
aXBwaW5nIHBvc2l0aW9uIGluaXQKMjAxNy0xMC0yNyAgNzoxNjoyMyAxMzk5OTQwNDczNTUxMzYg
W05vdGVdIFdTUkVQOiB3c3JlcF9sb2FkKCk6IGxvYWRpbmcgcHJvdmlkZXIgbGlicmFyeSAnL3Vz
ci9saWI2NC9nYWxlcmEtMy9saWJnYWxlcmFfc21tLnNvJwoyMDE3LTEwLTI3ICA3OjE2OjIzIDEz
OTk5NDA0NzM1NTEzNiBbTm90ZV0gV1NSRVA6IHdzcmVwX2xvYWQoKTogR2FsZXJhIDMuMjEocjg2
Nzg1MzgpIGJ5IENvZGVyc2hpcCBPeSA8aW5mb0Bjb2RlcnNoaXAuY29tPiBsb2FkZWQgc3VjY2Vz
c2Z1bGx5LgoyMDE3LTEwLTI3ICA3OjE2OjIzIDEzOTk5NDA0NzM1NTEzNiBbTm90ZV0gV1NSRVA6
IENSQy0zMkM6IHVzaW5nIGhhcmR3YXJlIGFjY2VsZXJhdGlvbi4KMjAxNy0xMC0yNyAgNzoxNjoy
MyAxMzk5OTQwNDczNTUxMzYgW05vdGVdIFdTUkVQOiBGb3VuZCBzYXZlZCBzdGF0ZTogOWI1NmI3
NmMtYmE1YS0xMWU3LWI0Y2ItY2JiYTBiMDg3ODY4OjQyMCwgc2FmZV90b19ib290c3JhcDogMAoy
MDE3LTEwLTI3ICA3OjE2OjIzIDEzOTk5NDA0NzM1NTEzNiBbTm90ZV0gV1NSRVA6IFBhc3Npbmcg
Y29uZmlnIHRvIEdDUzogYmFzZV9kaXIgPSAvdmFyL2xpYi9teXNxbC87IGJhc2VfaG9zdCA9IDEw
LjI0NC45MC44ODsgYmFzZV9wb3J0ID0gNDU2NzsgY2VydC5sb2dfY29uZmxpY3RzID0gbm87IGRl
YnVnID0gbm87IGV2cy5hdXRvX2V2aWN0ID0gMDsgZXZzLmRlbGF5X21hcmdpbiA9IFBUMVM7IGV2
cy5kZWxheWVkX2tlZXBfcGVyaW9kID0gUFQzMFM7IGV2cy5pbmFjdGl2ZV9jaGVja19wZXJpb2Qg
PSBQVDAuNVM7IGV2cy5pbmFjdGl2ZV90aW1lb3V0ID0gUFQxNVM7IGV2cy5qb2luX3JldHJhbnNf
cGVyaW9kID0gUFQxUzsgZXZzLm1heF9pbnN0YWxsX3RpbWVvdXRzID0gMzsgZXZzLnNlbmRfd2lu
ZG93ID0gNDsgZXZzLnN0YXRzX3JlcG9ydF9wZXJpb2QgPSBQVDFNOyBldnMuc3VzcGVjdF90aW1l
b3V0ID0gUFQ1UzsgZXZzLnVzZXJfc2VuZF93aW5kb3cgPSAyOyBldnMudmlld19mb3JnZXRfdGlt
ZW91dCA9IFBUMjRIOyBnY2FjaGUuZGlyID0gL3Zhci9saWIvbXlzcWwvOyBnY2FjaGUua2VlcF9w
YWdlc19zaXplID0gMDsgZ2NhY2hlLm1lbV9zaXplID0gMDsgZ2NhY2hlLm5hbWUgPSAvdmFyL2xp
Yi9teXNxbC8vZ2FsZXJhLmNhY2hlOyBnY2FjaGUucGFnZV9zaXplID0gMTI4TTsgZ2NhY2hlLnJl
Y292ZXIgPSBubzsgZ2NhY2hlLnNpemUgPSAxMjhNOyBnY29tbS50aHJlYWRfcHJpbyA9IDsgZ2Nz
LmZjX2RlYnVnID0gMDsgZ2NzLmZjX2ZhY3RvciA9IDEuMDsgZ2NzLmZjX2xpbWl0ID0gMTY7IGdj
cy5mY19tYXN0ZXJfc2xhdmUgPSBubzsgZ2NzLm1heF9wYWNrZXRfc2l6ZSA9IDY0NTAwOyBnY3Mu
bWF4X3Rocm90dGxlID0gMC4yNTsgZ2NzLnJlY3ZfcV9oYXJkX2xpbWl0ID0gOTIyMzM3MjAzNjg1
NDc3NTgwNzsgZ2NzLnJlY3ZfcV9zb2Z0X2xpbWl0ID0gMC4yNTsgZ2NzLnN5bmNfZG9ub3IgPSBu
bzsgZ21jYXN0LnNlZ21lbnQgPSAwOyBnbWNhc3QudmVyc2lvbiA9IDA7IHBjLmFubm91bmNlX3Rp
bWVvdXQgPSBQVDNTOyBwYy5jaGVja3N1bSA9IGZhbHNlOyBwYy5pCg==
```

**Comment by ducky-hong** [comment]
https://github.com/fluent/fluent-bit/pull/427 might be a related issue.

**Comment by edsiper** [comment]
@epcim  thanks for the test case.

as @ducky-hong said, this was related to the problem that in_tail stopped monitoring a file when a NULL byte was found. Based on @ducky-hong work, we published a fix for the case that will be available on Fluent Bit v0.13.0

Closing this issue since it has been solved on GIT Master
